geonode/base/views.py

```diff
@@ -301,23 +302,28 @@ def get_results(self, context):
 
 class ThesaurusAvailable(autocomplete.Select2QuerySetView):
     def get_queryset(self):
+
         tid = self.request.GET.get("sysid")
-        lang = self.request.GET.get("lang")
-        qs_local = []
-        qs_non_local = []
-        for key in ThesaurusKeyword.objects.filter(thesaurus_id=tid):
-            label = ThesaurusKeywordLabel.objects.filter(keyword=key).filter(lang=lang)
-            if self.q:
-                label = label.filter(label__icontains=self.q)
-            if label.exists():
-                qs_local.append(label.get())
-            else:
-                if self.q in key.alt_label:
-                    qs_non_local.append(key)
-                elif not self.q:
-                    qs_non_local.append(key)
-
-        return qs_non_local + qs_local
+
+        # help to get real language string
+        requested_lang = get_language_from_request(self.request)
```

Comment:
If you have time, can you please add a couple of tests on the APi?

Reply:
i updated this mechanism again to be more clean in my opinion.

Reply:
i am going to add a testcase for different language codes here.